pac4j-core/src/main/java/org/pac4j/core/client/ClientsFactoryDIImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.core.client;
+
+import java.util.List;
+
+import org.pac4j.core.credentials.Credentials;
+import org.pac4j.core.profile.UserProfile;
+
+/**
+ * A Spring-initializable factory to build clients.
+ * 
+ * @author Daniel Hankins
+ * @since 1.8.0
+ *
+ */
+public class ClientsFactoryDIImpl implements ClientsFactory {
+
+    /**
+     * The list of clients to be initialized by dependency injection and
+     * returned by build()
+     */
+    private Clients clientList;
+
+    /**
+     * Constructor that Spring can use to inject the client list to be returned
+     * by build()
+     * 
+     * @param clientList
+     */
+    public ClientsFactoryDIImpl(List<Client> clientList) {
```

Comment:
I see three items final could apply to: the class, the instance variable, and the constructor parameter. I must be missing something, because I don't see why you'd want to declare any of those three final, with the possible exception of the constructor argument.

Why would you want to make the class non-extensible or immutable? I can envision scenarios in which you'd want to make a class that can be initialized by dependency injection, then have some of the clients removed because they don't fit the context you're using the factory in.
The same scenario applies to the member variable.

I mean, you might want to construct a second list of clients from the first meeting some criteria, then assign it to the instance variable. Not possible if it's final.

I can see making the constructor argument final, as that would prevent you from inadvertently reassigning the argument variable to something else. Is that what you meant?